mp/src/materialsystem/stdshaders/pbr_dx9.cpp

```diff
@@ -0,0 +1,524 @@
+            if (bHasFlashlight)
+            {
+                pShaderShadow->EnableTexture(SHADER_SAMPLER5, true);        // flashlight cookie
+                pShaderShadow->EnableSRGBRead(SHADER_SAMPLER5, true);       // flashlight cookie is srgb
+                pShaderShadow->EnableTexture(SHADER_SAMPLER6, true);        // shadow depth map
+                pShaderShadow->SetShadowDepthFiltering(SHADER_SAMPLER6);    // set shadow depth filtering
+                pShaderShadow->EnableSRGBRead(SHADER_SAMPLER6, false);      // shadow depth map is not srgb
+                pShaderShadow->EnableTexture(SHADER_SAMPLER7, true);        // random rotation map
+            }
+
+            // if lightmaps are used, enable the sampler
+            if (bLightmapped)
+            {
+                pShaderShadow->EnableTexture(SHADER_SAMPLER8, true); // lightmap
+            }
+
+            // enabling srgb writing
+            // see common_ps_fxc.h line 349
+            // ps2b shaders and up write srgb
+            pShaderShadow->EnableSRGBWrite(true);
+
+            // set up vertex format
+            if (bLightmapped) // brush
+            {
+                // we only need the position and surface normal
+                unsigned int flags = VERTEX_POSITION | VERTEX_NORMAL;
+                // we need three texcoords, all in the default float2 size
+                pShaderShadow->VertexShaderVertexFormat(flags, 3, 0, 0);
+            }
+            else // model
+            {
+                // we need the position, surface normal, and vertex compression format
+                unsigned int flags = VERTEX_POSITION | VERTEX_NORMAL | VERTEX_FORMAT_COMPRESSED;
+                // we only need one texcoord, in the default float2 size
+                pShaderShadow->VertexShaderVertexFormat(flags, 1, 0, 0);
+            }
+
+            // setting up static vertex shader
+            DECLARE_STATIC_VERTEX_SHADER(pbr_vs20);
+            SET_STATIC_VERTEX_SHADER_COMBO(LIGHTMAPPED, bLightmapped);
+            SET_STATIC_VERTEX_SHADER(pbr_vs20);
+
+            // setting up static pixel shader
+            DECLARE_STATIC_PIXEL_SHADER(pbr_ps30);
+            SET_STATIC_PIXEL_SHADER_COMBO(FLASHLIGHT, bHasFlashlight);
+            SET_STATIC_PIXEL_SHADER_COMBO(FLASHLIGHTDEPTHFILTERMODE, nShadowFilterMode);
+            SET_STATIC_PIXEL_SHADER_COMBO(LIGHTMAPPED, bLightmapped);
+            SET_STATIC_PIXEL_SHADER(pbr_ps30);
+
+            // setting up fog
+            DefaultFog(); // this is probably correct
+
+            // enable alpha writes all the time so that we have them for underwater stuff
+            pShaderShadow->EnableAlphaWrites(bFullyOpaque);
+        }
+        else // not snapshotting - begin dynamic state
+        {
+            // setting up albedo texture
+            if (bHasAlbedo)
+            {
+                this->BindTexture(SHADER_SAMPLER0, info.Albedo, info.BaseTextureFrame);
```

Comment:
Drop the `this->`, it's inconsistently used across the file, and IMO isn't needed.

Reply:
I thought I had already removed those, apparently not. I'll remove them now